source/databases-collections.txt

```diff
@@ -0,0 +1,275 @@
+collections in the database and their associated metadata.
+
+The following example calls the ``mongoc_database_find_collections_with_opts()`` function and iterates over
+the cursor to print the results:
+
+.. code-block:: c
+   char *str;
+   const bson_t *doc;
+
+   cursor = mongoc_database_find_collections_with_opts (database, NULL, NULL);
+   
+   while (mongoc_cursor_next (cursor, &doc)) {
+        str = bson_as_canonical_extended_json(doc, NULL);
+        printf("Collection: %s\n", str);
+        bson_free(str);
+    }
+
+To query for only the names of the collections in the database, call the
+``mongoc_database_get_collection_names_with_opts()`` function as follows:
+
+.. code-block:: c
+
+   char **strv;
+   bson_error_t error;
+   unsigned i;
+
+   if ((strv = mongoc_database_get_collection_names_with_opts (database, NULL, &error))) {
+     for (i = 0; strv[i]; i++)
+       printf ("%s\n", strv[i]);
+
+     bson_strfreev (strv);
+   } else {
+     fprintf (sterr, "Command failed: %s\n", error.message);
+   }
+
+.. TODO: Uncomment when cursors page is created
+.. For more information about iterating over a cursor, see :ref:`c-cursors`. 
+
+Delete a Collection
+-------------------
+
+You can delete a collection from the database by using the ``mongoc_collection_drop()``
+function.
+
+The following example deletes the ``test_collection`` collection:
+
+.. code-block:: c
+
+   mongoc_collection_t *collection = mongoc_database_get_collection (database, "test_collection");
+   monogc_collection_drop (collection, NULL);
+
+.. warning:: Dropping a Collection Deletes All Data in the Collection
+
+   Dropping a collection from your database permanently deletes all
+   documents and all indexes within that collection.
+
+   Drop a collection only if the data in it is no longer needed.
+
+.. _c-config-read-write:
+
+Configure Read and Write Operations
+-----------------------------------
+
+You can control how the driver routes read operations by setting a **read preference**.
+You can also control options for how the driver waits for acknowledgment of
+read and write operations on a replica set by setting a **read concern** and a
+**write concern**.
+
+By default, databases inherit these settings from the ``mongoc_client_t`` instance,
+and collections inherit them from the database. However, you can change these
+settings on your database or collection by using one of the following functions:
+
+- For ``mongoc_database_t`` instances, use the ``mongoc_database_set_read_prefs()``, 
+  ``mongoc_database_set_read_concern()``, and ``mongoc_database_set_write_concern()`` functions
+  to set the read preferences, read concern, and write concern for the database.
+- For ``mongoc_collection_t`` instances, use the use the ``mongoc_collection_set_read_prefs()``, 
+  ``mongoc_collection_set_read_concern()``, and ``mongoc_collection_set_write_concern()`` functions
+  to set the read preferences, read concern, and write concern for the collection.
+
```

Comment:
S: instead of including three code examples that set collection settings, I think having one example that sets database settings and one that sets collection settings makes more sense. If you break up the list above, as mentioned in my last suggestion, you could add the database example after the list of database functions and same thing with collection